src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -2039,7 +2033,6 @@ static int section_setattro(NPySecObj* self, PyObject* pyname, PyObject* value)
     }
     PyObject* rv;
     int err = 0;
-    auto _pyname_tracker = nb::borrow(pyname);  // keep refcount+1 during use
```

Comment:
The commit message was not far from it. But lmk if you meant a PR

Reply:
We only do squash merges. Therefore, after merging the terms PR and commit become synonymous. It also implies that each PR can only have one theme if one wants to avoid commits on `master` that don't contain the word `and` or `but`.

Likely being too pedantic, but my gut feeling would say separate PR to have reduce the size of this PR and keep topics separate. Since, we've spent a lot of time discussing the refcounting in one function changed. Meaning the PR without these changes is already non-trivial and might be revisited.